Agents/FenlandTrajectoryAgent/agent/kgutils/utils.py

```diff
@@ -0,0 +1,168 @@
+# The purpose of this module is to provide settings and functions relevant to
+# both 1) instantiating and also 2) retrieving time series objects to/from KG
+# ===============================================================================
+import os
+import re
+import psycopg2
+import requests
+from pathlib import Path
+
+from configobj import ConfigObj
+from pathlib import Path
+
+from agent.utils.stack_configs import DB_URL, DB_USER, DB_PASSWORD, SPARQL_QUERY_ENDPOINT, SPARQL_UPDATE_ENDPOINT, DATABASE_NAME, NAMESPACE
+# Define format of time series time entries: Year-Month-Day T hour:minute:second Z
+FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+
+PROPERTIES_FILE = os.path.abspath(os.path.join(Path(__file__).parent.parent, 'resources', 'properties.properties'))
+# Define PREFIXES for SPARQL queries (WITHOUT trailing '<' and '>')
+PREFIXES = {
+    # Namespace for this example data
+
+    # Namespaces for used ontologies
+    'ex': 'http://www.theworldavatar.com/kb/fenlandtrajectory/',
+    'rdf': 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
+    'rdfs': 'http://www.w3.org/2000/01/rdf-schema#',
+    'ts': 'http://www.theworldavatar.com/kb/ontotimeseries/',
+    'xsd': 'http://www.w3.org/2001/XMLSchema#',
+    'geolit': 'http://www.bigdata.com/rdf/geospatial/literals/v1',
+    'geo': 'http://www.bigdata.com/rdf/geospatial#',
+    'ontodevice': 'https://www.theworldavatar.com/kg/ontodevice.owl'
+}
+
+
+def validate_and_initialize_configs():
+    """
+    Validates the configurations for database and SPARQL endpoints received from stack_configs.
+    """
+    # Validate configurations received from stack_configs.py
+    if not DB_URL or not DB_USER or not DB_PASSWORD:
+        raise ValueError("Database configurations are incomplete.")
+    if not SPARQL_QUERY_ENDPOINT or not SPARQL_UPDATE_ENDPOINT:
+        raise ValueError("SPARQL endpoint configurations are incomplete.")
+
+    print("Configurations validated successfully.")
+
+def create_sparql_prefix(abbreviation):
+    """
+        Constructs proper SPARQL Prefix String for given namespace abbreviation.
+
+        Arguments:
+            abbreviation - namespace abbreviation to construct SPARQL PREFIX string for.
+
+        Returns:
+            SPARQL query prefix string in the form "PREFIX ns: <full IRI>".
+    """
+
+    # Define global scope for global variables
+    global PREFIXES
+
+    # Raise key error if given namespace abbreviation has not been specified
+    if abbreviation not in PREFIXES.keys():
+        raise KeyError('Prefix: "' + abbreviation + '" has not been specified')
+
+    # Get full IRI from pre-specified prefixes dictionary
+    iri = PREFIXES[abbreviation]
+
+    if not iri.startswith('<'):
+        iri = '<' + iri
+    if not iri.endswith('>'):
+        iri = iri + '>'
+
+    return 'PREFIX ' + abbreviation + ': ' + iri + ' '
+
+
+def set_mapbox_apikey():
```

Comment:
I think we no longer set the mapbox_key in the index.html, please check [here](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/Deploy/stacks/dynamic/stack-manager#example---including-a-visualisation) on how to specify the mapbox_username and api